lib/portlayer/exec/container.go

```diff
@@ -393,16 +393,24 @@ func infraContainers(ctx context.Context, sess *session.Session) ([]mo.VirtualMa
 
 // find the childVM for this resource pool by name
 func childVM(ctx context.Context, sess *session.Session, name string) (*vm.VirtualMachine, error) {
-	searchIndex := object.NewSearchIndex(sess.Client.Client)
-	child, err := searchIndex.FindChild(ctx, VCHConfig.ResourcePool.Reference(), name)
```

Comment:
What if the vm path is renamed by vsphere? for example for vm test, vsphere might use test_1 as the folder name, if it found duplicate.
And do you know if that works for vSAN?